### PR TITLE
Run tests on Python 2.7

### PR DESCRIPTION
**Pull request by La0**
* Use mock for both python2 & python3
* Modify some cli code to run osf
* Update mock imports
* Add python 2.7 in travis

### CODE REVIEW

**Comment by betatim** [comment]
Oh wow, I had just started working on this in #52 should I close that one?

**Comment by betatim** [comment]
I think the answer to my question is yes :)

**Comment by La0** [comment]
@betatim Sorry i just saw your own PR when i pushed mine.

**Comment by betatim** [comment]
This fixes #48 and #46 .

File: osfclient/__main__.py [hunk 1]
```diff
@@ -54,26 +61,17 @@ def main():
...
-        )
+    list_parser = _add_subparser('list', list.__doc__, aliases=['ls'])
     list_parser.set_defaults(func=list_)
 
     # Upload a single file
-    upload_parser = subparsers.add_parser(
-        'upload', description=upload.__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter
-        )
+    upload_parser = _add_subparser('upload', upload.__doc__, aliases=['ls'])
```
Comment (betatim) [inline osfclient/__main__.py]: spurious alias here

**Comment by La0** [comment]
@betatim Python 2 support is here ;)

File: osfclient/models/storage.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 from .file import File
 from ..utils import norm_remote_path
 
+if six.PY2:
+    class FileExistsError(OSError):
+        """
+        Exception raised when a file already exists
+        Standard in Python 3
+        """
+
```
Comment (betatim) [inline osfclient/models/storage.py]: We don't need a `pass` here?

Comment (betatim) [inline osfclient/models/storage.py]: Answered my own question by trying it out. Seems with the docstring you don't need a `pass`. Every day is a school day!

**Comment by betatim** [comment]
Merging!

**Pull request merged**